library/src/data/index.js

```diff
@@ -41,20 +41,265 @@ window.CoNDeT.data = {
     },
 
     saveToFile: function (data) {
-      var element = document.createElement('a');
+      var element = document.createElement("a");
       element.setAttribute(
-        'href',
-        'data:text/plain;charset=utf-8,' +
+        "href",
+        "data:text/plain;charset=utf-8," +
           encodeURIComponent(JSON.stringify(data))
       );
-      element.setAttribute('download', 'CoNDeT.json');
+      element.setAttribute("download", "CoNDeT.json");
 
-      element.style.display = 'none';
+      element.style.display = "none";
       document.body.appendChild(element); //Required on firefox
 
       element.click();
 
       element.remove();
     },
   },
+
+  StateModifier: (function () {
+    constructor = function (state) {
+      this.stateManager = state;
+      var self = this;
+      state.subscribe(function (newState) {
+        self.state = newState;
+      });
+    };
+
+    moveTable = function (tableId, newCoordinates) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+      editTable.coordinates = {
+        x: newCoordinates.x,
+        y: newCoordinates.y,
+      };
+
+      this.stateManager.setState(newState);
+    };
+
+    createTable = function (newData) {
+      var newState = clone(this.state);
+      newState.push(newData);
+      this.stateManager.setState(newState);
+    };
+
+    editName = function (tableId, name) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      if (name != null) {
+        editTable.name = name;
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    editClass = function (tableId, className) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      if (className != null) {
+        editTable.className = className;
+      }
+
+      this.stateManager.setState(newState);
+    };
+
+    editCondition = function (tableId, conditions) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      replaceInArray(editTable.columns.conditions, conditions);
+
+      this.stateManager.setState(newState);
+    };
+
+    editDecision = function (tableId, decisions) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      replaceInArray(editTable.columns.decisions, decisions);
+
+      this.stateManager.setState(newState);
+    };
+
+    editCell = function (tableId, cell) {
+      var newState = clone(this.state);
+      var editTable = getTableWithId(newState, tableId);
+
+      if (cell != null) {
+        for (var i = 0; i < editTable.rows.length; i++) {
+          if (editTable.rows[i].rowId == cell.rowId) {
+            editTable.rows[i].conditions = cell.conditions;
+            editTable.rows[i].decisions = cell.decisions;
+          }
+        }
+      }
+
+      this.stateManager.setState(newState);
+    };
```

Comment:
Tutaj będziemy chcieli podawać: `tableId`, `rowId`, `type`, `index`, `value`. Gdzie:
* `type` to `conditions` lub `decisions`
* `index` indeks decyzji/warunku  
(reszta chyba oczywista)
Można zastosować podobną strategię jak przy `removeColumnWithId` i ewentualnie rozdzielić na osobne funkcje dla `conditions` i `decisions` (wtedy bez parametru `type`)